lib/image.js

```diff
@@ -12,37 +12,53 @@
 
 const bindings = require('./bindings')
 const Image = module.exports = bindings.Image
+const http = require("http")
 
-/**
- * Src setter.
- *
- *  - convert data uri to `Buffer`
- *
- * @param {String|Buffer} val filename, buffer, data uri
- * @api public
- */
+Object.defineProperty(Image.prototype, 'src', {
+  /**
+   * src setter. Valid values:
+   *  * `data:` URI
+   *  * Local file path
+   *  * HTTP or HTTPS URL
+   *  * Buffer containing image data (i.e. not a `data:` URI stored in a Buffer)
+   *
+   * @param {String|Buffer} val filename, buffer, data URI, URL
+   * @api public
+   */
+  set(val) {
+    if (typeof val === 'string') {
+      if (/^\s*data:/.test(val)) { // data: URI
+        const commaI = val.indexOf(',')
+        // 'base64' must come before the comma
+        const isBase64 = val.lastIndexOf('base64', commaI)
+        val = val.slice(commaI + 1)
+        this.source = Buffer.from(val, isBase64 ? 'base64' : 'utf8')
+      } else if (/^\s*http/.test(val)) { // remote URL
+        const onerror = typeof this.onerror === 'function' ? this.onerror : () => {}
```

Comment:
I _think_ that it's not too uncommon to do something like:

```js
img.src = '...'
img.onerror = ...
```

How about something like:

```js
const onerror = (err) => typeof this.onerror === 'function' && this.onerror(err)
```

Reply:
Good point, we could at the very least just log an error? Sort of like when a Promise rejects and you haven't caught it:

```
> const a = Promise.reject(new Error('Something went wrong'))
undefined
> (node:20870) UnhandledPromiseRejectionWarning: Error: Something went wrong
    at repl:1:26
    at Script.runInThisContext (vm.js:91:20)
    at REPLServer.defaultEval (repl.js:317:29)
    at bound (domain.js:396:14)
    at REPLServer.runBound [as eval] (domain.js:409:12)
    at REPLServer.onLine (repl.js:615:10)
    at REPLServer.emit (events.js:187:15)
    at REPLServer.EventEmitter.emit (domain.js:442:20)
    at REPLServer.Interface._onLine (readline.js:290:10)
    at REPLServer.Interface._line (readline.js:638:8)
(node:20860) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
```

Reply:
Now that I think about it more, it might be a bad idea...

The `UnhandledPromiseRejectionWarning` can be silenced (`node --no-warnings`). There'd be no equivalent way for a user to silence an `onerror` console message. I also can't think of a stream or other event emitter that causes a warning or message if you don't listen to its `"error"` event.

Hopefully having the docs always show using the `onerror` handler is enough.

Reply:
Every event emitter, streams included, actually crashes the entire process if you emit an `error` and there is no listeners for it.

```text
➜  /tmp cat test.js 
const events = require('events')
const ee = new events.EventEmitter()

ee.emit('error', new Error('Hello, World!'))

setTimeout(() => console.log('Nothing happened...'), 2000)
➜  /tmp node test.js
events.js:160
      throw er; // Unhandled 'error' event
      ^

Error: Hello, World!
    at Object.<anonymous> (/private/tmp/test.js:4:18)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.runMain (module.js:604:10)
    at run (bootstrap_node.js:389:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:502:3
➜  /tmp 
```